### DISCUSSION TITLE
Designer SDK Runtime Dependencies

### DISCUSSION THREAD

**Discussion by amaitland** [dotnet/winforms#6757] [Q&A] [answered]
After reading [State of the Windows Forms Designer for .NET Applications](https://devblogs.microsoft.com/dotnet/state-of-the-windows-forms-designer-for-net-applications/) and seeing [Microsoft.WinForms.Designer.SDK](https://www.nuget.org/packages/Microsoft.WinForms.Designer.SDK) is a required dependency to support the new `OOP Designer` I've gone and installed the package. 

```xml
<PackageReference Include="Microsoft.WinForms.Designer.SDK" Version="1.1.0-prerelease-preview3.22076.5" />
```

The package adds numerous design time only dependencies which flow through into my example applications `bin` folder (runtime dependencies).  When ideally these would only be design (or compile) time dependencies.

Is it possible to mark the package as `PrivateAssets="All"`? Can this be the default? Can I avoid adding a reference to my `Nuget` packages?

What's the general guidance here? I cannot seem to find any articles that address the topic.

**Comment by KlausLoeffelmann** [comment, marked as answer]
We're working on them.
While I previous predicted the samples and docs should be ready around now, it'll be likely later in March.
Sorry for the delay.

**Reply by amaitland** [reply]
Thanks for the prompt reply. Greatly appreciated.

> While I previous predicted the samples and docs should be ready around now, it'll be likely later in March.

Before releasing the guide please take the time to review the **Nuget** package. I think it's important that the issues are resolved before the wider public starts using the package.

1. If at all possible make the package a [developmentDependency](https://docs.microsoft.com/en-us/nuget/reference/nuspec#developmentdependency). This would resolve my concerns raised above (no further guidance would be necessary).
2. Fix the `Project website` and `Source repository` links which both point to a non existent github repo (https://github.com/dotnet/winforms-designer)

Looking forward to the guide 👍

**Reply by amaitland** [reply]
Any update? I've checked the blog and no sign of a guide. 

Thanks.

**Reply by jamiehighfield** [reply]
See this discussion where there is access to an early sample project that may help with architecture etc.

https://github.com/dotnet/winforms/discussions/7073

**Reply by amaitland** [reply]
Thanks for the link, greatly appreciated 👍

**Comment by giduac** [comment]
Hello @KlausLoeffelmann,

Is it possible to provide some more info on when the SDK Docs (and possibly sample apps) will become available?

Cheers...,
